addons/website/models/website_visitor.py

```diff
@@ -0,0 +1,129 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import fields, models, api
+from odoo.http import request
+from datetime import datetime, timedelta
+from hashlib import md5
+
+
+class WebsiteVisitor(models.Model):
+    _name = 'website.visitor'
+    _description = 'Website Visitor'
+    _order = 'last_connection_date DESC'
+
+    name = fields.Char('Name', default='Website Visitor')
+    website_id = fields.Many2one('website', "Website", readonly=True)
+    partner_ids = fields.Many2many('res.partner')
+    partner_count = fields.Integer('# Partners', compute="_compute_partner_count")
+    create_date = fields.Datetime('First connexion date', readonly=True)
+    last_connection_date = fields.Datetime('Last Connection', help="Last page view date", index=True, readonly=True)
+    country_id = fields.Many2one('res.country', 'Country', readonly=True)
+    country_flag = fields.Binary(related="country_id.image", string="Country Flag")
+    lang_id = fields.Many2one('res.lang', string='Language', help="Language from the website when visitor has been created")
+    visit_count = fields.Integer('Number of visits', default=1, readonly=True, help="A new visit is considered if last connection was more than 8 hours ago.")
+    page_ids = fields.One2many('website.visitor.page', 'visitor_id', string='Visited Pages History')
+    page_view_count = fields.Integer('Page Views', compute="_compute_page_view_count")
+    unique_page_ids = fields.Many2many('website.page', string="Visited Pages", compute="_compute_unique_page_ids", store=True)
+    time_last_action = fields.Float('Time since last action', compute="_compute_time_last_action", help='Time since last page view in seconds')
+    is_connected = fields.Boolean('Is connected ?', compute='_compute_is_connected', help='A visitor is considered as connected if his last page view was within the last 5 minutes.')
+
+    @api.depends('partner_ids')
+    def _compute_partner_count(self):
+        for visitor in self:
+            visitor.partner_count = len(visitor.partner_ids)
+
+    @api.depends('page_ids')
+    def _compute_page_view_count(self):
+        for visitor in self:
+            visitor.page_view_count = len(visitor.page_ids)
+
+    @api.depends('page_ids')
+    def _compute_unique_page_ids(self):
+        for visitor in self:
+            unique_page_ids = list(set(visitor.page_ids.page_id.ids))
```

Comment:
I think a read_group on visitor.page, group_by visitor_ir, reading also page_id, could probably allow to compute both page_view_count and unique_page_ids in a single compute using a nice read group.